heatmap/src/heatmap.rs

```diff
@@ -196,40 +209,52 @@ impl Heatmap {
         self.summary.buckets()
     }
 
+    /// Returns the number of valid and active `Histogram` slices in the `Heatmap`
+    pub fn active_slices(&self) -> usize {
+        if Instant::now() - self.first_tick > self.span {
+            self.slices()
+        } else {
+            self.idx.load(Ordering::Relaxed) + 1
+        }
+    }
+
+    /// Returns the DateTime representation of when the `HeatMapi` was created
+    pub fn created_at(&self) -> DateTime {
+        self.create_at
+    }
+
     /// Increment a time-value pair by a specified count
     pub fn increment(&self, time: Instant, value: u64, count: u32) {
         self.tick(time);
 
-        let current = self.current.load(Ordering::Relaxed);
+        let curr_tick = self.curr_tick.load(Ordering::Relaxed);
+        let idx = self.idx.load(Ordering::Relaxed);
 
         // fast path when the time falls into the current time slice
-        if time >= self.slices[current].start.load(Ordering::Relaxed) {
+        if time >= curr_tick {
             let _ = self.summary.increment(value, count);
-            let _ = self.slices[current].histogram.increment(value, count);
+            let _ = self.histograms[idx].increment(value, count);
         }
 
-        // since it's before the current time slice, we need to record it into
-        // a past slice
+        // the duration belonged to a past histogram
```

Comment:
nit: the heatmap / histogram doesn't necessarily store durations. We could say the sample or timestamp belongs to a past histogram. We might be recording block io sizes or some count.

Reply:
good point. fixed.